### ISSUE TITLE
Compile error on Alpine (with musl)

### ISSUE THREAD

**Issue by tcdude** [panda3d/panda3d#875]
When trying to compile (with gcc) Panda3D on Alpine Linux, the build fails instantly on the first task `Building C++ object built/tmp/p3dtoolbase_composite1.o`. I independently tried this on ARM 64 bit and x86_64 to the same result.

The error message that is thrown:
```
In file included from dtool/src/dtoolbase/mutexPosixImpl.h:71,
                 from dtool/src/dtoolbase/mutexImpl.h:42,
                 from dtool/src/dtoolbase/neverFreeMemory.h:19,
                 from dtool/src/dtoolbase/deletedBufferChain.h:18,
                 from dtool/src/dtoolbase/deletedBufferChain.cxx:14,
                 from dtool/src/dtoolbase/p3dtoolbase_composite1.cxx:6:
dtool/src/dtoolbase/mutexPosixImpl.I: In constructor 'constexpr MutexPosixImpl::MutexPosixImpl()':
dtool/src/dtoolbase/mutexPosixImpl.I:19:1: error: temporary of non-literal type 'pthread_mutex_t' in a constant expression
   19 | }
      | ^
In file included from /usr/include/sys/types.h:57,
                 from dtool/src/dtoolbase/dtoolbase.h:168,
                 from dtool/src/dtoolbase/addHash.h:17,
                 from dtool/src/dtoolbase/addHash.cxx:14,
                 from dtool/src/dtoolbase/p3dtoolbase_composite1.cxx:1:
/usr/include/bits/alltypes.h:71:16: note: 'pthread_mutex_t' is not literal because:
   71 | typedef struct { union { int __i[10]; volatile int __vi[10]; volatile void *volatile __p[5]; } __u; } pthread_mutex_t;
      |                ^
/usr/include/bits/alltypes.h:71:96: note:   non-static data member 'pthread_mutex_t::__u' has non-literal type
   71 | typedef struct { union { int __i[10]; volatile int __vi[10]; volatile void *volatile __p[5]; } __u; } pthread_mutex_t;
      |                                                                                                ^~~
/usr/include/bits/alltypes.h:71:24: note: 'pthread_mutex_t::<unnamed union>' is not literal because:
   71 | typedef struct { union { int __i[10]; volatile int __vi[10]; volatile void *volatile __p[5]; } __u; } pthread_mutex_t;
      |                        ^
/usr/include/bits/alltypes.h:71:52: note:   non-static data member 'pthread_mutex_t::<unnamed union>::__vi' has volatile type
   71 | typedef struct { union { int __i[10]; volatile int __vi[10]; volatile void *volatile __p[5]; } __u; } pthread_mutex_t;
      |                                                    ^~~~
/usr/include/bits/alltypes.h:71:86: note:   non-static data member 'pthread_mutex_t::<unnamed union>::__p' has volatile type
   71 | typedef struct { union { int __i[10]; volatile int __vi[10]; volatile void *volatile __p[5]; } __u; } pthread_mutex_t;
      |                                                                                      ^~~
```

The systems I have tried this on:
**aarch64**
 * uname -a: `Linux pinephone 5.5.0 #1-postmarketos-allwinner SMP Wed Feb 26 12:39:14 UTC 2020 aarch64 Linux`
 * ldd --version: `musl libc (aarch64) / Version 1.1.24`

**x86_64**
 * uname -a: `Linux f88fb01cadd5 5.3.0-29-generic #31-Ubuntu SMP Fri Jan 17 17:27:26 UTC 2020 x86_64 Linux` (alpine linux with docker on Ubuntu Host)
 * ldd --version: `musl libc (x86_64) / Version 1.1.24`

**Note:**
    I successfully managed to compile Panda3D on the same device (Pinephone), running Manjaro ARM, the main difference there is that Alpine Linux uses `musl` instead of `glibc`

**Comment by rdb** [comment]
Should be fixed on 1.10 branch.

**Comment by tcdude** [comment]
I can confirm that this specific error is fixed now. I ran into another problem, which I'm gathering information about currently, before potentially opening a separate issue for it.
